payment_paypal/indico_payment_paypal/controllers.py

```diff
@@ -107,13 +119,28 @@ class RHPaypalSuccess(RHPaypalIPN):
     """Confirmation message after successful payment"""
 
     def _process(self):
+        # Force check on transaction
+        sleeps=0
+        registration = Registration.query.filter_by(uuid=self.token).first()
+
+        while ((not registration.is_paid) and (sleeps < 30)):
+            current_plugin.logger.warning("Waiting for transaction registration %s",sleeps)
+            time.sleep(1)
+            sleeps=sleeps+1
+            db.session.expire(registration)
+            registration = Registration.query.filter_by(uuid=self.token).first()
+
         flash(_('Your payment request has been processed.'), 'success')
-        return redirect(url_for('event_registration.display_regform', self.registration.locator.registrant))
+        redir_url=url_for('event_registration.display_regform', self.registration.locator.registrant)+"&utime="+str(time.time())
```

Comment:
This is not the appropriate way to pass parameters to a URL. you can simply pass them as named arguments of `url_for`:
```suggestion
        redir_url=url_for('event_registration.display_regform', self.registration.locator.registrant, utime=str(time.time()))
```

Reply:
done.